### PR TITLE
[MIN-339] Implement getUserUnderlyingBalancePerAsset RPC

### PR DESCRIPTION
**Pull request by dev-vadim**
**Description:**

Added RPC to return amount of underlying tokens user currently have in protocol.

**Changes in storage or API:**

Controller:
- getUserUnderlyingBalancePerAsset

Describe storage structure changes or API changes. Make sure to update the [API docs](https://minterestfinance.atlassian.net/wiki/spaces/MINTEREST/pages/134807557/Minterest+API).

**Checklist**

- [ ] Code is covered with tests
- [ ] Code is covered with benchmarks
- [ ] Fresh benchmark results included

### CODE REVIEW

**Review by lut1k — DISMISSED** [review]

File: runtime/src/tests/rpc_tests.rs [hunk 1]
```diff
@@ -803,6 +803,85 @@ fn get_user_total_collateral_rpc_should_work() {
 		})
 }
 
+#[test]
+fn test_get_user_underlying_balance_per_asset_rpc() {
```
Comment (kurotych) [inline runtime/src/tests/rpc_tests.rs:807]: Could you add a short scenario of this test?

Comment (dev-vadim) [reply runtime/src/tests/rpc_tests.rs:820]: Done

**Review by lut1k — DISMISSED** [review]

File: runtime/src/tests/rpc_tests.rs [hunk 2]
```diff
@@ -803,6 +803,98 @@ fn get_user_total_collateral_rpc_should_work() {
...
+
+			assert_ok!(MinterestProtocol::redeem(alice(), ETH));
+			// Check that after redeem user ETH balance increased by a value that was previously returned by
+			// get_user_underlying_balance_per_asset
+			assert_eq!(
+				Currencies::free_balance(ETH, &ALICE::get()),
+				eth_balance_after_repay + balance_converted_to_underlying_after_borrowing
+			);
+		})
+}
```
Comment (kurotych) [inline runtime/src/tests/rpc_tests.rs:896]: Your test a bit confused me. I suggest to make it simpler.
```suggestion
#[test]
fn test_get_user_underlying_balance_per_asset_rpc() {
	ExtBuilder::default()
		.pool_initial(DOT)
		.pool_initial(ETH)
		.build()
		.execute_with(|| {
			assert_ok!(MinterestProtocol::deposit_underlying(bob(), ETH, dollars(90_000)));
			// Alice deposited ALL ETH tokens to protocol
			assert_ok!(MinterestProtocol::deposit_underlying(alice(), ETH, dollars(100_000)));

			assert_eq!(
				get_user_underlying_balance_per_asset_rpc(ALICE::get(), ETH),
				Some(BalanceInfo {
					amount: dollars(100_000)
				})
			);

			// Bob makes a borrow to update exchange rate
			assert_ok!(MinterestProtocol::enable_is_collateral(bob(), ETH));
			assert_ok!(MinterestProtocol::borrow(bob(), ETH, dollars(40_000)));

			// Alice Underlying balance hasn`t changed in the block when first borrow occurred
			assert_eq!(
				get_user_underlying_balance_per_asset_rpc(ALICE::get(), ETH),
				Some(BalanceInfo {
					amount: dollars(100_000)
				})
			);

			// Skip some blocks to accrue interest. Bob repay his borrow
			System::set_block_number(100);
			assert_ok!(MinterestProtocol::repay_all(bob(), ETH));

			// 3554127423600000 - this is interest that alice earn for depositing
			let alice_balance_after_borrowing = dollars(100_000) + 3554127423600000;
			assert_eq!(
				get_user_underlying_balance_per_asset_rpc(ALICE::get(), ETH),
				Some(BalanceInfo {
					amount: alice_balance_after_borrowing
				})
			);

			// Converts ALL mTokens into a specified quantity of the underlying asset
			assert_ok!(MinterestProtocol::redeem(alice(), ETH));
			// Check that value got by rpc and value after redeem the same
			assert_eq!(
				Currencies::free_balance(ETH, &ALICE::get()),
				alice_balance_after_borrowing
			);
		})
}
```

Comment (dev-vadim) [reply runtime/src/tests/rpc_tests.rs]: Changed test to your version but removed pool_init for DOT and repay_all(bob(), ETH)

**Review by kurotych — APPROVED** [review]

**Review by lut1k — APPROVED** [review]

**Pull request merged**